CiscoVM/CiscoVM-1.0.0/ciscovm_helpers.py

```
@@ -53,44 +54,31 @@ def _generate_headers(self, event_type: str = CHECK_EVENT_TYPE) -> dict:
             "Authorization": f"Bearer {self.auth_token}",
         }
 
-    def response_handler(func):
-        """Handle any request errors and logging them."""
+    def with_retries(func):
+        """Adds retry logic to request"""
         @functools.wraps(func)
         def wrap(self, *args, **kwargs):
-            msg = ""
             for attempt in range(self.RETRY_STRATEGY["total"]):
-                try:
-                    response = func(self, *args, **kwargs)
-                    if response.status_code == 200:
-                        logging.debug(f"Data was sent to {self.full_url}")
-                        return True, msg
-                    elif response.status_code in self.RETRY_STRATEGY["status_forcelist"]:
-                        # Retry request with backoff delay
-                        delay = self.RETRY_STRATEGY["backoff_factor"] * (2 ** attempt)
-                        msg = f"Status code: {response.status_code}"
-                        logging.debug(f"{msg}. Retrying in {delay} seconds...")
-                        time.sleep(delay)
-                        continue
-                    else:
-                        msg = (f"Status code: {response.status_code} "
-                               f"Response msg: {response.text}")
-                        logging.debug(msg)
-                        return False, msg
-                except Exception as e:
-                    msg = str(e)
-                    break
-
-            logging.debug(msg)
-            return False, msg
+                response = func(self, *args, **kwargs)
+                if response.status_code in self.RETRY_STRATEGY["status_forcelist"]:
+                    # Retry request with backoff delay
+                    delay = self.RETRY_STRATEGY["backoff_factor"] * (2 ** attempt)
+                    msg = f"Status code: {response.status_code}"
+                    logging.debug(f"{msg}. Retrying in {delay} seconds...")
```

Lets remove msg var

---

Actually, should we remove that? Basically, we suppress error here fore retry purpose, so logging is the only way to know if the retry happend and why.